### DISCUSSION TITLE
Direct data workbook tagged ?data not found

### DISCUSSION THREAD

**Discussion by Karaka** [climateinteractive/SDEverywhere#157] [Q&A] [answered]
hi,  grateful for any thoughts...anything jump out?

- Have set-up SDE on my M1 mac, using VS Code Editor and its integrated terminal to test and run models.
- Entering the command: sde generate --genc --spec {filename} {model name}   
- And keep getting the "Direct data workbook tagged ?data not found" message when using some of the test (and my own) models ...all of which include a direct data variable (including the directdata test model)
- terminal screenshot attached showing command and output

![PastedGraphic-4](https://user-images.githubusercontent.com/7798808/139690108-e277566f-1517-4ac2-a1b2-4f8cb9fe33a9.png)

Bit more info:
- SDE generate --list {model} command works fine in all cases
- am running the commands from the src folder of SDEverywhere (and get the same behaviour if i run the commands from within the relevant model folder where the data.xlsx file is located)
- my local folder structure attached
- have linked both SDE and antlr4-vensim folders to npm and installed the SDE dependencies as per the wiki instructions
- Node v16.9.1 installed
![PastedGraphic-3](https://user-images.githubusercontent.com/7798808/139689351-a524065c-6b7c-4b7b-b8c8-98708ce497a1.png)

**Comment by chrispcampbell** [comment, marked as answer]
@Karaka It looks like the issue is in the path to the "spec.json" file.  In the screenshot above, the command includes `--spec ...models/directdata_spec.json`, but it should instead be `--spec ...models/directdata/directdata_spec.json` (need to include the `directdata` dir).

This highlights an issue that `sde generate` is not failing fast if the spec file can't be found; if it was working correctly, it would've been easier to pinpoint the issue right away.  I filed an issue for that in #158.

**Reply by Karaka** [reply]
Thank you @chrispcampbell ... sure enough...much appreciated.  Now generates the C code fine.
